azurerm/internal/services/network/resource_arm_application_gateway.go

```diff
@@ -1001,8 +1001,9 @@ func resourceArmApplicationGateway() *schema.Resource {
 						},
 
 						"key_vault_secret_id": {
-							Type:     schema.TypeString,
-							Optional: true,
+							Type:         schema.TypeString,
+							Optional:     true,
+							ValidateFunc: azure.ValidateKeyVaultChildId,
```

Comment:
@katbyte i am quite sure this break the autorenew feature of the application gateway

```
After Application Gateway is configured to use Key Vault certificates, its instances retrieve the certificate from Key Vault and install them locally for SSL termination. The instances also poll Key Vault at 24-hour intervals to retrieve a renewed version of the certificate, if it exists. If an updated certificate is found, the SSL certificate currently associated with the HTTPS listener is automatically rotated.
```

With your validator you need to specify an exact version of the secret so the renew will not work.
You just need to point to the secret without the version.

Reply:
I agree. The current tf code wants 3 parts : vaulturl/certificate name/hash

This adds it but app gateway never sees new versions of the cert. In order to have this it's only needs 2 part: vaulturl/certificate name.

Reply:
@francescopersico since this PR's been merged, can you open a new issue to track that? Thanks!

Reply:
It's already exists #[6188](https://github.com/terraform-providers/terraform-provider-azurerm/issues/6188)